tests-js/other/LoanOpeningsBorrowOrTradeFromPool.test.js

```diff
@@ -0,0 +1,207 @@
+			console.log("sovrynBeforeSUSDBalance", sovrynBeforeSUSDBalance.toString());
+
+			const sovrynBeforeRBTCBalance = await RBTC.balanceOf(sovryn.address);
+			console.log("sovrynBeforeRBTCBalance", sovrynBeforeRBTCBalance.toString());
+
+			const loanTokenSent = hunEth;
+
+			await SUSD.mint(sovryn.address, loanTokenSent, { from: accounts[0] });
+
+			const sovrynSwap = await sovryn.sovrynSwapContractRegistryAddress();
+			// console.log('sovryn seap contract registry address is ',sovrynSwap)
+			// addressOf = sovrynSwap.addressOf(sovrynSwap.address)
+			console.log("returned address is ", sovrynSwap);
+			const swapsI = await sovryn.swapsImpl();
+			console.log("swaps impl is ", swapsI);
+			const collateralTokenSent = await sovryn.getRequiredCollateral(
+				SUSD.address,
+				RBTC.address,
+				loanTokenSent,
+				new BN(50).mul(oneEth),
+				false
+			);
+			console.log("required collateral:", collateralTokenSent.div(oneEth).toString());
+
+			await RBTC.mint(sovryn.address, collateralTokenSent, { from: accounts[0] });
+
+			console.log("loanTokenSent", loanTokenSent.toString());
+			console.log("collateralTokenSent", collateralTokenSent.toString());
+
+			const tx = await sovryn.borrowOrTradeFromPool(
+				await LinkDaiMarginParamsId(), //loanParamsId
+				"0x0", // loanId
+				false, // isTorqueLoan,
+				hunEth, // initialMargin
+				[
+					accounts[2], // lender
+					accounts[1], // borrower
+					accounts[1], // receiver
+					constants.ZERO_ADDRESS, // manager
+				],
+				[
+					new BN(5).mul(oneEth), // newRate (5%)
+					loanTokenSent, // newPrincipal
+					0, // torqueInterest
+					loanTokenSent, // loanTokenSent
+					collateralTokenSent, // collateralTokenSent
+				],
+				"0x", // loanDataBytes
+				{ from: accounts[1] }
+			);
+
+			const sovrynAfterSUSDBalance = await SUSD.balanceOf(sovryn.address);
+			console.log("sovrynAfterSUSDBalance", sovrynAfterSUSDBalance.toString());
+
+			const sovrynAfterRBTCBalance = await RBTC.balanceOf(sovryn.address);
+			console.log("sovrynAfterRBTCBalance", sovrynAfterRBTCBalance.toString());
+
+			const decode = decodeLogs(tx.receipt.rawLogs, LoanOpeningsEvents, "Trade");
+			const tradeEvent = decode[0].args;
+
+			const interestForPosition = loanTokenSent
+				.mul(new BN(5).mul(oneEth))
+				.div(hunEth)
+				.div(new BN(365))
+				.mul(new BN(2419200))
+				.div(new BN(86400));
+			console.log("interestForPosition", interestForPosition.toString());
+
+			// expectedPositionSize = collateralTokenSent + ((loanTokenSent - interestForPosition) * tradeEvent["entryPrice"] // 1e18)
+			const expectedPositionSize = loanTokenSent
+				.sub(interestForPosition)
+				.mul(new BN(tradeEvent["entryPrice"]))
+				.div(oneEth)
+				.add(collateralTokenSent);
+
+			// ignore differences in least significant digits due to rounding error
+			expect(Math.abs(expectedPositionSize.sub(new BN(tradeEvent["positionSize"])).toNumber()) < 100).to.be.true;
```

Comment:
```suggestion
			expect(expectedPositionSize.divn(100)).to.be.bignumber.equal(new BN(tradeEvent["positionSize"]).divn(100));
```

Reply:
i see the previous not fixed code

Reply:
It doesn't pass if we update code like this.

Reply:
it should work with the updated codebase from bzx
ok. leaving as is for now as this is obviously a rounding problem